### PR TITLE
Test/105 review page accessibility tests

### LINKED ISSUE #105: Add accessibility tests for the review page using `jest-axe`
The review page has no automated accessibility checks. Add `jest-axe` tests that verify the page doesn't have common accessibility violations (contrast, missing labels, improper heading structure).

**Relevant files:**
- `frontend/src/pages/__tests__/ReviewPage.test.tsx`

**Estimated effort:** 4–6 hours

### PR DESCRIPTION
**Pull request by thisiswhale**
## Summary
Adds automated accessibility coverage for the review page using `jest-axe`, checking each of its three conditionally-rendered states (polling, failed, complete) plus the stacked-error-banner case, since a violation could exist in one state and not another.

## Issue
Closes #105

## Changes
- Register `jest-axe`'s `toHaveNoViolations` matcher globally in `frontend/src/test/setup.ts`
- Add `frontend/src/pages/__tests__/ReviewPage.test.tsx`:
  - mocks `react-router-dom` (`useParams`/`useNavigate`) and `useReviewStatus`/`apiClient`, following the existing `vi.mock` pattern from `ProfileForm.test.tsx`
  - `jest-axe` checks for the polling, failed, and complete states
  - `jest-axe` check for the polling-error + fetch-error banners rendered together

## Testing
- [x] Unit tests pass (`make test-unit`)
- [ ] Integration tests pass (`make test-integration`)
- [ ] Linter passes (`make lint`)
- [ ] Type checker passes (`make typecheck`)
- [x] New/updated tests cover the changes

`make test-unit` and `make lint`/`make typecheck` are Python-only and don't cover this frontend-only change. Baselined pre-existing failures on `main` before starting (all in files this PR never touches, confirmed disjoint):
- `make lint`: 182 ruff errors (`agent/`, `ingestion/`, `rag/`, `tests/unit/`)
- `make format` (check mode): 52 files not black-formatted
- `make typecheck`: 5 mypy errors (missing stubs for `PyPDF2`, `jose`, `passlib`, `rank_bm25`, plus a numpy stub issue)
- `make test-unit`: 53 of 428 pytest tests failing
- Frontend (`npx vitest run`, not covered by `make`): `ReviewSection.test.tsx` has 1 pre-existing failure; `ProfileForm.test.tsx` can't run (missing `@testing-library/user-event` dep)

New `ReviewPage.test.tsx`: 5/5 passing.

**To manually verify:**
1. `cd frontend && npm test -- ReviewPage` — should report 5 passing tests, 0 failures.
2. To see the states directly: `make run` (or `cd frontend && npm run dev`), log into the app with one of the seed accounts in `docs/SETUP.md` (e.g. `user1@example.com` / `password1`), start a review from the dashboard, and open `/reviews/<reviewId>`. You'll see the polling state immediately, then either the failed or complete state once processing finishes.
3. Optional: with a browser a11y extension (e.g. axe DevTools) open on that page, re-check each state — this is the same check the new tests automate, so it should also report zero violations.

## Screenshots / Demo
N/A — test-only change, no UI changes.

## Notes for Reviewers
PLAN.md (in this branch) documents the state-by-state approach and the risks/edge cases considered before writing the tests.

### CODE REVIEW

File: frontend/src/pages/__tests__/ReviewPage.test.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    expect(await axe(container)).toHaveNoViolations()
+  })
+
+  it('has no accessibility violations in the complete state', async () => {
+    mockUseReviewStatus.mockReturnValue({ review: completeReview, isPolling: false, error: null })
+    mockGetReview.mockResolvedValue(completeReview)
+
+    const { container } = render(<ReviewPage />)
+
+    await screen.findByText('Portfolio Review')
```
Comment (rafayet-git) [inline frontend/src/pages/__tests__/ReviewPage.test.tsx]: I would recommend changing this to something more obvious that the score has arrived, such as "Overall Score".

**Review by rafayet-git — APPROVED** [review]
Overall it looks good to me. just a small change I would recommend. All test pass and works fine.
